switools/swisignature.py

```diff
@@ -163,23 +162,25 @@ def signSwiHandler( args ):
       signSwiAll( workDir, swi, signingCertFile, rootCaFile, signatureFile, signingKeyFile )
       print( 'SWI/X file %s successfully signed and verified.' % swi )
 
-def runCmd( cmd, workDir ):
+def runCmd( cmd, workDir=None ):
    try:
-      subprocess.check_call( cmd.split(" "), cwd=workDir )
+      subprocess.check_call( cmd, cwd=workDir )
    except subprocess.CalledProcessError:
       return False
    return True
 
 def insertSignature( swi, sigFileName, workDir ):
-   cmd = "zip -q -0 -X %s %s" % ( os.path.abspath( swi ), sigFileName )
+   cmd = [ "zip", "-q", "-0", "-X", os.path.abspath( swi ) ]
+   cmd = cmd + sigFileName.split( " " )
```

Comment:
It would be better to pass `sigFileName` as a list rather than joining then splitting the names.